sendgrid.go

```diff
@@ -26,19 +26,38 @@ type SGClient struct {
 }
 
 // NewSendGridClient will return a new SGClient.
-func NewSendGridClient(apiUser, apiPwd string) *SGClient {
+func NewSendGridClient(apiUserOrKey string, apiKey ...string) *SGClient {
 	apiMail := "https://api.sendgrid.com/api/mail.send.json?"
-	return &SGClient{
-		apiUser: apiUser,
-		apiPwd:  apiPwd,
-		APIMail: apiMail,
+
+	var Client *SGClient
+
+	// Detect if given username and password or an api key
+	if len(apiKey) == 1 {
+		// Username and password
+		Client = &SGClient{
+			apiUser: apiUserOrKey,
+			apiPwd:  apiKey[0],
+			APIMail: apiMail,
+		}
+	} else if len(apiKey) == 0 {
+		// API key
+		Client = &SGClient{
+			apiPwd:  apiUserOrKey,
+			APIMail: apiMail,
+		}
+	} else {
+		panic("Arguments error in NewSendGridClient")
```

Comment:
Hey @eddiezane, in general you want to avoid `panic` in a public API's because you are basically deciding for the end-user code that the app should shut down.  If we can, it'd be better to return an `error` instead.

Example: 
- https://github.com/Shopify/sarama/blob/master/client.go#L96